### PR TITLE
TDAP057-Mark-Done-Functionality

### PR DESCRIPTION
**Pull request by amit-ib**
Added mark done functionality

### CODE REVIEW

File: src/components/todo/TodoItem.tsx [hunk 1]
```diff
@@ -72,7 +89,16 @@ const TodoItem = ({ todoItem, setLoading, setTost }: Props) => {
...
-        <Button label="Mark Done" className="link" onClick={doneTaskHandeler} />
+        {/* todo.status === ToDoStatus.COMPLETED
+          ? ToDoStatus.PENDING
+          : ToDoStatus.COMPLETED */}
+        <Button
+          label={
+            todo.status === ToDoStatus.COMPLETED ? "Not Done" : "Mark Done"
+          }
+          className="link"
+          onClick={doneTaskHandeler}
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:100]: Is to mark it done, or it's a toggle.

File: src/components/todo/TodoItem.tsx [hunk 2]
```diff
@@ -48,8 +48,26 @@ const TodoItem = ({ todoItem, setLoading, setTost }: Props) => {
...
+    //dispatch(markDoneTodoAction(todoItem));
+    let updateData = {
+      title: todo.title,
+      status:
+        todo.status === ToDoStatus.COMPLETED
+          ? ToDoStatus.PENDING
+          : ToDoStatus.COMPLETED,
+      dueDate: todo.dueDate,
+      category: todo.category,
+    };
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:61]: I guess this to update only the status correct?

**Review by akash-dhande-ib — DISMISSED** [review]

File: src/components/todo/TodoItem.tsx (hunk 1 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx:104]: this is toggle

File: src/components/todo/TodoItem.tsx [hunk 3]
```diff
@@ -48,8 +48,22 @@ const TodoItem = ({ todoItem, setLoading, setTost }: Props) => {
...
 
-  const doneTaskHandeler = () => {
-    dispatch(markDoneTodoAction(todoItem));
+  const doneTaskHandeler = async () => {
+    let updateData = {
+      status:
+        todo.status === ToDoStatus.COMPLETED
+          ? ToDoStatus.PENDING
+          : ToDoStatus.COMPLETED,
+    };
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:57]:
```suggestion
    let updatedData = {
      ...todoItem,
      status: ToDoStatus.COMPLETED,
    };
```

File: src/components/todo/TodoItem.tsx [hunk 4]
```diff
@@ -59,20 +73,23 @@ const TodoItem = ({ todoItem, setLoading, setTost }: Props) => {
...
-
   return (
     <div className={`list-item`} id={String(todoItem.id)}>
       <div
         className={`list-item-text 
           ${todoItem.status === ToDoStatus.COMPLETED ? "task-done" : ""}
         `}
       >
+        <span
+          onClick={doneTaskHandeler}
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:84]: toggleStatusHandler

File: src/components/todo/TodoItem.tsx [hunk 5]
```diff
@@ -59,20 +73,23 @@ const TodoItem = ({ todoItem, setLoading, setTost }: Props) => {
...
+          className={
+            todo.status === ToDoStatus.COMPLETED ? "checked" : "unchecked"
+          }
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:87]:
```suggestion
          className={`todo-status-icon  todo-status-icon-${todo.status === ToDoStatus.COMPLETED ? "checked" : "unchecked"`
          }
```

File: src/components/todo/TodoItem.tsx (hunk 3 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: Implemented as discussed

File: src/components/todo/TodoItem.tsx (hunk 4 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: updated as discussed

**Review by akash-dhande-ib — APPROVED** [review]

**Pull request merged**